bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -144,65 +161,73 @@ private void longPoll(BoschHttpClient httpClient, String subscriptionId) {
-                }
+                // NOTE: This handler runs inside the HTTP thread, so we schedule the response handling in a new thread
+                // because the HTTP thread is terminated after the timeout expires.
+                scheduler.execute(() -> longPolling.onLongPollComplete(httpClient, subscriptionId, result,
+                        this.getContentAsString()));
             }
         });
     }
 
-    private void onLongPollResponse(BoschHttpClient httpClient, String subscriptionId, String content) {
+    /**
+     * This is the handler for responses of long poll requests.
+     * 
+     * @param httpClient HTTP client which received the response
+     * @param subscriptionId Id of subscription the response is for
+     * @param result Complete result of the response
+     * @param content Content of the response
+     */
+    private void onLongPollComplete(BoschHttpClient httpClient, String subscriptionId, @Nullable Result result,
+            String content) {
         // Check if thing is still online
         if (this.aborted) {
             logger.debug("Canceling long polling for subscription id {} because it was aborted", subscriptionId);
             return;
         }
 
-        logger.debug("Long poll response: {}", content);
+        Throwable failure = result != null ? result.getFailure() : null;
+        if (failure != null) {
+            if (failure instanceof ExecutionException) {
+                if (failure.getCause() instanceof AbortLongPolling) {
+                    logger.debug("Canceling long polling for subscription id {} because it was aborted",
+                            subscriptionId);
+                } else {
+                    this.handleFailure.accept(new LongPollingFailedException(
+                            "Unexpected exception during long polling request", failure));
+                }
+            } else {
+                this.handleFailure.accept(
+                        new LongPollingFailedException("Unexpected exception during long polling request", failure));
+            }
+        } else {
+            logger.debug("Long poll response: {}", content);
 
-        String nextSubscriptionId = subscriptionId;
+            String nextSubscriptionId = subscriptionId;
 
-        LongPollResult longPollResult = gson.fromJson(content, LongPollResult.class);
-        if (longPollResult != null && longPollResult.result != null) {
-            this.handleResult.accept(longPollResult);
-        } else {
-            logger.warn("Long poll response contained no results: {}", content);
+            LongPollResult longPollResult = gson.fromJson(content, LongPollResult.class);
+            if (longPollResult != null && longPollResult.result != null) {
+                this.handleResult.accept(longPollResult);
+            } else {
+                logger.warn("Long poll response contained no results: {}", content);
```

Comment:
All ``logger.warn`` should be Info or Debug level, because you code can now handle long poll problems and work after a re-subscript or reconnect again.